File: src/Domain/Entities/Notification.cs
using Domain.Enums;

namespace Domain.Entities;

public class Notification
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public NotificationType Type { get; set; }
    public string Message { get; set; } = string.Empty;
    public Guid? TaskId { get; set; }
    public string? TaskTitle { get; set; }
    public Guid ActorId { get; set; }
    public string ActorName { get; set; } = string.Empty;
    public bool IsRead { get; set; }
    public DateTime CreatedAt { get; set; }

    public User User { get; set; } = null!;
}

File: src/Web/src/types/notification.ts
export interface NotificationDto {
  id: string;
  type: string;
  message: string;
  taskId?: string;
  taskTitle?: string;
  actorId: string;
  actorName: string;
  isRead: boolean;
  createdAt: string;
}

export interface NotificationPagedResult {
  items: NotificationDto[];
  totalCount: number;
  page: number;
  pageSize: number;
}
